macOS: fails to build

### Describe the bug

```
Checking system → [✔]
Running command → scons -j8 │
────────────────────────────┘

scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
Building compilation database compile_commands.json
Retrieved `selfdrive/modeld/models/dmonitoring_model_tinygrad.pkl' from cache
Retrieved `selfdrive/modeld/models/driving_policy_tinygrad.pkl' from cache
Retrieved `selfdrive/modeld/models/driving_vision_tinygrad.pkl' from cache
  [LINK] tools/cabana/_cabana
ld: warning: ignoring duplicate libraries: '-lcapnp', '-lkj', 'cereal/libcereal.a', 'cereal/libsocketmaster.a', 'msgq_repo/libmsgq.a', 'msgq_repo/libvisionipc.a'
Undefined symbols for architecture arm64:
  "_CFBooleanGetTypeID", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
  "_CFBooleanGetValue", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
  "_CFDataGetBytes", referenced from:
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
  "_CFDataGetLength", referenced from:
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
  "_CFDataGetTypeID", referenced from:
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
  "_CFDictionaryCreateMutable", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_CFDictionarySetValue", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_CFGetTypeID", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_CFNumberCreate", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_CFNumberGetTypeID", referenced from:
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_interface in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_CFNumberGetValue", referenced from:
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_interface in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_CFRelease", referenced from:
      _darwin_close in libusb-1.0.a[10](darwin_usb.o)
      _darwin_close in libusb-1.0.a[10](darwin_usb.o)
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_CFRetain", referenced from:
      _darwin_open in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopAddSource", referenced from:
      _darwin_open in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_claim_interface in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopGetCurrent", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopRemoveSource", referenced from:
      _darwin_close in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_release_interface in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopRun", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopSourceCreate", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopSourceSignal", referenced from:
      _darwin_exit in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopStop", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_CFRunLoopWakeUp", referenced from:
      _darwin_exit in libusb-1.0.a[10](darwin_usb.o)
  "_CFUUIDGetConstantUUIDWithBytes", referenced from:
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_CFUUIDGetUUIDBytes", referenced from:
      _darwin_device_from_service in libusb-1.0.a[10](darwin_usb.o)
      _darwin_claim_interface in libusb-1.0.a[10](darwin_usb.o)
  "_IOCreatePlugInInterfaceForService", referenced from:
      _darwin_device_from_service in libusb-1.0.a[10](darwin_usb.o)
      _darwin_claim_interface in libusb-1.0.a[10](darwin_usb.o)
  "_IOIteratorNext", referenced from:
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_attached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_attached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_clear_iterator in libusb-1.0.a[10](darwin_usb.o)
      _darwin_clear_iterator in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_IOKitWaitQuiet", referenced from:
      _darwin_hotplug_poll in libusb-1.0.a[10](darwin_usb.o)
  [LINK] tools/cabana/tests/test_cabana
  "_IONotificationPortCreate", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_IONotificationPortDestroy", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_IONotificationPortGetRunLoopSource", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_IOObjectRelease", referenced from:
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
      _darwin_exit in libusb-1.0.a[10](darwin_usb.o)
      _darwin_kernel_driver_active in libusb-1.0.a[10](darwin_usb.o)
      _darwin_kernel_driver_active in libusb-1.0.a[10](darwin_usb.o)
      _darwin_destroy_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_IOObjectRetain", referenced from:
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
  "_IORegistryEntryCreateCFProperty", referenced from:
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_IORegistryEntryGetChildEntry", referenced from:
      _darwin_kernel_driver_active in libusb-1.0.a[10](darwin_usb.o)
  "_IORegistryEntryGetParentEntry", referenced from:
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
  "_IOServiceAddMatchingNotification", referenced from:
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
  "_IOServiceAuthorize", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
  "_IOServiceGetMatchingService", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_IOServiceGetMatchingServices", referenced from:
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
  "_IOServiceMatching", referenced from:
      _darwin_init in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_SecTaskCopyValueForEntitlement", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
  "_SecTaskCreateFromSelf", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
  "___CFConstantStringClassReference", referenced from:
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
       in libusb-1.0.a[10](darwin_usb.o)
      ...
  "_kCFAllocatorDefault", referenced from:
      _darwin_detach_kernel_driver in libusb-1.0.a[10](darwin_usb.o)
      _darwin_devices_detached in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_cached_device in libusb-1.0.a[10](darwin_usb.o)
      _darwin_get_interface in libusb-1.0.a[10](darwin_usb.o)
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_kCFAllocatorSystemDefault", referenced from:
      _get_interface_interface in libusb-1.0.a[10](darwin_usb.o)
      _get_device_interface in libusb-1.0.a[10](darwin_usb.o)
  "_kCFRunLoopCommonModes", referenced from:
      _darwin_open in libusb-1.0.a[10](darwin_usb.o)
  "_kCFRunLoopDefaultMode", referenced from:
      _darwin_close in libusb-1.0.a[10](darwin_usb.o)
      _darwin_event_thread_main in libusb-1.0.a[10](darwin_usb.o)
      _darwin_release_interface in libusb-1.0.a[10](darwin_usb.o)
      _darwin_claim_interface in libusb-1.0.a[10](darwin_usb.o)
  "_kCFTypeDictionaryKeyCallBacks", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
  "_kCFTypeDictionaryValueCallBacks", referenced from:
      _usb_find_interface_matching_location in libusb-1.0.a[10](darwin_usb.o)
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  [LINK] tools/replay/tests/test_replay
ld: warning: ignoring duplicate libraries: '-lcapnp', '-lkj', 'cereal/libcereal.a', 'cereal/libsocketmaster.a', 'msgq_repo/libmsgq.a', 'msgq_repo/libvisionipc.a'
scons: *** [tools/cabana/_cabana] Error 1
scons: building terminated because of errors.

```

### OS Version

26.3.1 (25D2128)

### openpilot version or commit

master

### Additional info

_No response_

--- a/tools/cabana/SConscript
+++ b/tools/cabana/SConscript
@@ -69,7 +69,7 @@ base_frameworks = qt_env['FRAMEWORKS']
 base_libs = [common, messaging, cereal, visionipc, 'm', 'pthread'] + qt_env["LIBS"]
 
 if arch == "Darwin":
-  base_frameworks.append('QtCharts')
+  base_frameworks += ['QtCharts', 'CoreFoundation', 'IOKit', 'Security']
 else:
   base_libs.append('Qt5Charts')
 

--- a/tools/cabana/cabana
+++ b/tools/cabana/cabana
@@ -25,6 +25,18 @@ install_qt() {
   fi
 }
 
+get_job_count() {
+  if command -v nproc &> /dev/null; then
+    nproc
+  elif [[ "$(uname)" == "Darwin" ]]; then
+    sysctl -n hw.ncpu
+  elif command -v getconf &> /dev/null; then
+    getconf _NPROCESSORS_ONLN
+  else
+    echo 2
+  fi
+}
+
 # Install Qt if not found
 if ! command -v qmake &> /dev/null; then
   echo "Qt not found, installing dependencies..."
@@ -33,6 +45,6 @@ fi
 
 # Build _cabana
 cd "$ROOT"
-scons -j"$(nproc)" tools/cabana/_cabana
+scons -j"$(get_job_count)" tools/cabana/_cabana
 
 exec "$DIR/_cabana" "$@"

--- a/tools/cabana/cabana
+++ b/tools/cabana/cabana
@@ -25,18 +25,6 @@ install_qt() {
   fi
 }
 
-get_job_count() {
-  if command -v nproc &> /dev/null; then
-    nproc
-  elif [[ "$(uname)" == "Darwin" ]]; then
-    sysctl -n hw.ncpu
-  elif command -v getconf &> /dev/null; then
-    getconf _NPROCESSORS_ONLN
-  else
-    echo 2
-  fi
-}
-
 # Install Qt if not found
 if ! command -v qmake &> /dev/null; then
   echo "Qt not found, installing dependencies..."
@@ -45,6 +33,6 @@ fi
 
 # Build _cabana
 cd "$ROOT"
-scons -j"$(get_job_count)" tools/cabana/_cabana
+scons -j4 tools/cabana/_cabana
 
 exec "$DIR/_cabana" "$@"

